org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -583,23 +583,23 @@ public synchronized void stop() {
 		FileBuffers.getTextFileBufferManager().removeFileBufferListener(fileBufferListener);
 	}
 
-	public @Nullable CompletableFuture<LanguageServerWrapper> connect(@Nullable IDocument document, IFile file) {
+	public CompletableFuture<@Nullable LanguageServerWrapper> connect(@Nullable IDocument document, IFile file) {
```

Comment:
I have already realized this `return null`, and I agree that this seems better, but it is a breaking change for any consumer that was handling the null case.

Personally I am okay with it, I know we have code that relies on null, but I also know how to refactor. However, I it might break other projects.

Therefore I would argue we should let it be. @mickaelistria , any opinions?

Reply:
I have no hard feelings about this change. I can revert it if required.